04.ls/ls.rb

```diff
@@ -2,33 +2,109 @@
 # frozen_string_literal: true
 
 require 'optparse'
+require 'etc'
 
 COL_MAX = 3
 PADDING = 2
+FILE_TYPE_TO_CHARACTER = {
+  'file' => '-',
+  'directory' => 'd',
+  'characterSpecial' => 'c',
+  'blockSpecial' => 'b',
+  'fifo' => 'p',
+  'link' => 'l',
+  'socket' => 's',
+  'unknown' => '?'
+}.freeze
 
 def main
-  options = { r: false }
+  options = { l: false }
   opt = OptionParser.new
-  opt.on('-r') { options[:r] = true }
+  opt.on('-l') { options[:l] = true }
   argv = opt.parse(ARGV)
 
-  path = argv[0] || '.'
-  FileTest.directory?(path) or return
-  file_names = Dir.children(path).sort
-  display_file_names(file_names, options)
+  path = argv[0] || './'
+  corrected_path = path.sub(%r{([^/])\z}, '\1/') # pathの文字列が/で終わらない場合、文字列の末尾に/を追加する
```

Comment:
```suggestion
  corrected_path = path.end_with?('/') ? path : "#{path}/"
```

ってするとコメントがいらなくなりそうです